content/information-distribution/node.md

```diff
@@ -44,81 +44,132 @@ This ping **must** take the form of a Matrix timeline event of the
+at `node.example.com` to use Matrix identifiers ending with `:example.com`.
 
-In order to find the effective FQDN and port to reach a node at, client
+In order to find the effective host name and port to reach a node at, client
 implementations **must** implement the server discovery through `.well-known`
 URI logic, as [described in the Matrix
 specifications](https://matrix.org/docs/spec/client_server/r0.4.0.html#server-discovery).
 
 {{% notice note %}}
 If the server discovery leads to a final `IGNORE` instruction (as specified in
 the link above), then client implementations **must** use the node's server name
-as the FQDN the node can be reached at.
+as the host name the node can be reached at.
 {{% /notice %}}
 
 In the event of an entry node that's currently being used becoming unreachable,
-clients implementations **should** use the map generated from the room aliases
-list to provide its user(s) with alternative entry nodes and aliases to use.
-Client implementations **can** use received pings in order to filter out the
-inactive nodes from this list.
+clients implementations **should** use their list of nodes (either hardcoded by
+the implementor or computed from the list of users in the federation) to provide
+its user(s) with alternative entry nodes to use.
 
-### Joining a federation
+#### Joining a federation
 
-Once both an entry node and a room alias have been selected as the one to use in
+Client implementations **must** provide users with the room ID of at least one
+federation, and allow users to input custom values for the room ID of the
+federation to join.
+
+When reaching a node in order to join a federation, client implementations
+**must** try to retrieve the `/.well-kown/informo/info` file of the node (while
+following 30x redirects). If the file doesn't exist, or doesn't explicitly state
+that the node can be used as an entry node for the given federation, then client
+implementations **must** give up and try another node.
+
+Client implementations **should** store the result of the attempt at retrieving
+the `/.well-kown/informo/info` file of a node in a cache.
+
+Once both an entry node and a room ID have been selected as the one to use in
 order to join a federation, whether this selection originated from an explicit
 action by a user or, either partially or entirely, from a client
 implementation's inner mechanisms, client implementations **must** send a [join
 request](https://matrix.org/docs/spec/client_server/r0.4.0.html#post-matrix-client-r0-join-roomidoralias)
-to the selected node using the selected alias.
+to the selected node using the federation's room ID.
+
+In order to optimise the likelihood of success of the join request, client
+implementations **should** provide at least 3 nodes to join the federation
+through using `server_name` query parameters. Client implementations don't need
+to make sure that these nodes are entry nodes in this case.
```

Comment:
`Client implementations don't need to make sure that these nodes are entry nodes in this case.`

This is a bit ambiguous: you can understand that client implementations provides entry nodes that they are not sure if they are real entry nodes. I'd rephrase it like:
`Those nodes are expected to be entry nodes, and client implementations **can** try to join those nodes without verifying if they are effectively entry nodes.`

Reply:
Thing is, these nodes **aren't** expected to be entry nodes. All we care about is that they are in the room, and the only (indirect) interaction a client can generate with them is if the entry node the client is talking to isn't in the room, then that entry node is going to ask that other node to make it join it. Once it's done the client won't generate any other interaction with this other node, either direct or indirect (unless the other node is an entry node and the client eventually starts using it, or a similar scenario happens).

Reply:
Fixed in https://github.com/Informo/specs/pull/45/commits/e070a7792f07b22e58e3426a62573e1cc4baa2bf